examples/docker-ops/README.md

```diff
@@ -0,0 +1,127 @@
+Chef Workstation (and DK) commands thru the official Chef Docker images. This 
+eliminates the need to install the Workstation on your local development 
+machine and allows you to quickly and easily switch between new and previous 
+versions of Chef. This is important to be able to perform upgrade tasks to 
+cookbooks, and also test new versions of the software to ensure compatibility
+with your existing Chef code.
+
+## Before You Start
+
+### Assumptions
+
+* Are using Mac OSX or Linux (Windows Powershell coming soon)
+* Have an understanding of how to use the `bash_profile` standard, more here:
+https://stackoverflow.com/questions/8967843/how-do-i-create-a-bash-alias
+* Have Docker installed and running and are able to spin up containers and
+connect to the Docker registry
+* Are willing to contribute back to this repo with any new cool alias 
+corrections or functionality :)
+
+### Tested Versions
+
+* ChefDK/Workstation 1.5 | [`1.5 - thru - Chef Workstation Latest`]
+* Automate + Chef Infra Server | [`20201020140427`]
+* Docker for Mac | [`Docker version 19.03.8, build afacb8b`]
+
+## Copy the Embedded code into your bash profile file 
+(see link in assumptions for the location on your system)
+
+1. Copy this block of code into your profile file:
+    ```shell
+    # Chef Docker Aliases
+    func_chef () {
+      kitchen_driver="kitchen-docker"
+      image="${1}"
+      shift
+      args=("${@}")
+      if [[ " ${args[@]} " =~ " kitchen " ]]; then
+        command="gem install $kitchen_driver; $args;"
+      else
+        command="$args;"
+      fi
+      docker run --rm -i -w=/ws \
+        -v ~/.aws:/root/.aws \
+        -v ~/.gitconfig:/root/.gitconfig \
+        -v ~/.chef-workstation/cache/cookbooks/:/root/.chef-workstation/cache/cookbooks \
+        -v ~/.chef:/root/.chef \
+        -v $PWD:/ws \
+        -v /var/run/docker.sock:/var/run/docker.sock \
+        -v /usr/local/bin/docker:/usr/local/bin/docker \
+        --network host \
+        -e CHEF_LICENSE=accept \
+        $image \
+        /bin/bash -c "$command"
+    }
+    chef_12 () {
+      image="chef/chefdk:1.5"
+      local args=("$@")
+      func_chef "$image" "${args[@]}"
+    }
+    chef_13 () {
+      image="chef/chefdk:2.5.3"
+      local args=("$@")
+      func_chef "$image" "${args[@]}"
+    }
+    chef_14 () {
+      image="chef/chefdk:3.13.6"
+      local args=("$@")
+      func_chef "$image" "${args[@]}"
+    }
+    chef_15 () {
+      image="chef/chefworkstation:0.16.31"
+      local args=("$@")
+      func_chef "$image" "${args[@]}"
+    }
+    chef_16 () {
+      image="chef/chefworkstation:stable"
+      local args=("$@")
+      func_chef "$image" "${args[@]}"
+    }
+    ```
+1. Source the profile, on my Mac, using zshrc, I type `source ~/.zshrc`
+1. Run commands like this `chef_<version> [chef|knife|kitchen] arg1 arg2 etc....`:
+    ```shell
+    # Example - knife node list, ChefDK 3.12.10
+    > chef_14 knife node list
```

Comment:
Should this be `chefdk_3.12 knife node list`?  In your comment it has chefDK 3.12.10, but in the command it has chef_14

Reply:
kind of confusing IMO that we are calling chefdk 3.12.10, chef_14

Reply:
yeah it's a good point, I was trying to tie the dk/workstation to the version of Chef Client it was used for, if it makes more sense to alias the versions of the dk/workstation I can do that as well, maybe have multiple names for the alias (as a matter of fact, I'll do that and update the PR right now :) )

Reply:
Nice!  The other thought I had was that we want to ensure users understand they can run whatever version of chef-client, no matter what dk or workstation version they are using.  Sometimes customers think these two items are tied together.  More food for thought vs. action item.

Reply:
yeah good point, I guess the idea here is "if you are developing for "x-version" of chef, use this image", that way users can switch between different versions when needed (like when supporting older versions of legacy cookbooks for whatever reason, while also developing for newer versions)